server/conn.go

```diff
@@ -245,3 +260,16 @@ func (p *leaderProxy) handleRequest(msgID uint64, req *pdpb.Request) (*pdpb.Resp
 	}
 	return resp, errors.Trace(err)
 }
+
+func isUnexpectedConnError(err error) bool {
+	if err == nil {
+		return false
+	}
+	if errors.Cause(err) == io.EOF {
+		return false
+	}
+	if strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
maybe you can check the err is net.Error instead. 
It's better not to check the error message directly.